common/src/main/java/org/opensearch/ml/common/connector/AbstractConnector.java

```diff
@@ -65,11 +65,18 @@ public abstract class AbstractConnector implements Connector {
     protected User owner;
     @Setter
     protected AccessMode access;
+    @Setter
     protected Instant createdTime;
+    @Setter
     protected Instant lastUpdateTime;
     @Setter
     protected ConnectorClientConfig connectorClientConfig;
 
+    public AbstractConnector() {
+        this.createdTime = Instant.now();
```

Comment:
Will this method be used when retrieve an old connector ? The old connector has no create time and last update time. If this method used when retrieve old connector, it will assign `Instant.now()` to these two time fields, that's not expected.

Reply:
This constructor (AbstractConnector) will be invoked on the creation of (HTTPConnector) https://github.com/opensearch-project/ml-commons/blob/93d0429db5f7d32305362e985d0f9ffa908d5275/common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java#L45

and (AWSConnector) https://github.com/opensearch-project/ml-commons/blob/93d0429db5f7d32305362e985d0f9ffa908d5275/common/src/main/java/org/opensearch/ml/common/connector/AwsConnector.java#L30

These were suggestions from the initial post
https://github.com/opensearch-project/ml-commons/issues/2890#issue-2506354115

I might be misunderstanding your concern so please let me know

Reply:
@brianf-aws Assume this fix goes into 2.18.0.   If there was a connector created in 2.13.0, it will have null for both of those time fields.   The concern is what happens when that connector is parsed.

I believe the XContent parsing and Transport (Writeable) implementations will maintain the null, but this is something you should confirm.

Reply:
Hey @ylwu-amzn and @dbwiddis My newest commit c606ffadb77cb044881d9c2d97709ca6d13bce0b addresses this issue now upon indexing new connectors they will receive a time stamp while older connectors will have the same shape. Ive tested locally and via unit tests in [plugin/src/test/java/org/opensearch/ml/action/connector/UpdateConnectorTransportActionTests.java](https://github.com/opensearch-project/ml-commons/commit/8c006de118ae26db6ae1b52f3697a3aac6d43912#diff-2be9eb714846960c89f2cc23fd91699d39eee15cbfe030ef8e06b81ccb41032cL204)